heron/healthmgr/src/java/com/twitter/heron/healthmgr/sensors/BackPressureSensor.java

```diff
@@ -0,0 +1,93 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.twitter.heron.healthmgr.sensors;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import com.microsoft.dhalion.api.MetricsProvider;
+import com.microsoft.dhalion.metrics.ComponentMetrics;
+import com.microsoft.dhalion.metrics.InstanceMetrics;
+
+import com.twitter.heron.healthmgr.HealthPolicyConfig;
+import com.twitter.heron.healthmgr.common.PackingPlanProvider;
+import com.twitter.heron.healthmgr.common.TopologyProvider;
+
+public class BackPressureSensor extends BaseSensor {
+  private final MetricsProvider metricsProvider;
+  private final PackingPlanProvider packingPlanProvider;
+  private final TopologyProvider topologyProvider;
+
+  @Inject
+  public BackPressureSensor(PackingPlanProvider packingPlanProvider,
+                            TopologyProvider topologyProvider,
+                            HealthPolicyConfig policyConfig,
+                            MetricsProvider metricsProvider) {
+    super(policyConfig, MetricName.METRIC_BACK_PRESSURE.text());
+    this.packingPlanProvider = packingPlanProvider;
+    this.topologyProvider = topologyProvider;
+    this.metricsProvider = metricsProvider;
+  }
+
+  @Override
+  public Map<String, ComponentMetrics> get(String... components) {
+    return get();
+  }
+
+  /**
+   * Computes the average (millis/sec) back-pressure caused by instances in the configured window
+   *
+   * @return the average value
+   */
+  public Map<String, ComponentMetrics> get() {
+    Map<String, ComponentMetrics> result = new HashMap<>();
+
+    String[] boltComponents = topologyProvider.getBoltNames();
+    for (String boltComponent : boltComponents) {
+      String[] boltInstanceNames = packingPlanProvider.getBoltInstanceNames(boltComponent);
+
+      Duration duration = getDuration(BackPressureSensor.class.getSimpleName());
+      Map<String, InstanceMetrics> instanceMetrics = new HashMap<>();
+      for (String boltInstanceName : boltInstanceNames) {
+        String metric = this.metricName + boltInstanceName;
+        Map<String, ComponentMetrics> stmgrResult = metricsProvider.getComponentMetrics(
+            metric, duration, COMPONENT_STMGR);
+
+        HashMap<String, InstanceMetrics> streamManagerResult =
+            stmgrResult.get(COMPONENT_STMGR).getMetrics();
+
+        // since a bolt instance belongs to one stream manager, expect just one metrics
+        // manager instance in the result
+        InstanceMetrics stmgrInstanceResult = streamManagerResult.values().iterator().next();
+
+        double averageBp = stmgrInstanceResult.getMetricValueSum(metric) / duration.getSeconds();
+        averageBp = averageBp > 1000 ? 1000 : averageBp;
```

Comment:
`averageBP` is the number of millisec backpressure per second. Hence the value 1000. Sometimes Heron can double count the back pressure time and provide a value larger than 1000 (https://github.com/twitter/heron/issues/1753). This check partially corrects the error.